lib/bloc/backup/backup_bloc.dart

```diff
@@ -0,0 +1,144 @@
+    sharedPrefrences.then((preferences) {
+      _promptEnableBackupController
+          .add(preferences.getBool(BACKUP_DISABLED_PREFERENCES_KEY) ?? false);
+    });
+
+    _listenEnableBackupNotifications(sharedPrefrences);
+    _listenBackupPaths(breezLib);
+    _listenEnableBackupRequests(sharedPrefrences);
+    _listenBackupNowRequests(sharedPrefrences);
+    _listenRestoreRequests(breezLib);
+
+    _accountBloc.accountStream.listen((acc) {
+      _currentNodeId = acc.id;
+    });
+  }
+
+  void _listenEnableBackupNotifications(
+      Future<SharedPreferences> sharedPrefrences) {
+    _service.backupNotifications.listen((enabled) {
+      if (enabled) {
+        // Hide the backup disabled indicator
+        _backupDisabledController.add(false);
+      } else {
+        // Prompt to back up if not explicity disabled
+        sharedPrefrences.then((preferences) {
+          if (!(preferences.getBool(BACKUP_DISABLED_PREFERENCES_KEY) ??
+              false)) {
+            // Send a signal to prompt
+            _backupDisabledController.add(true);
+            _promptEnableBackupController.add(true);
+          }
+        });
+      }
+    });
+  }
+
+  void _listenEnableBackupRequests(Future<SharedPreferences> sharedPrefrences) {
+    _enableBackupController.stream.listen((enable) {
+      sharedPrefrences.then((preferences) {
+        preferences.setBool(BACKUP_DISABLED_PREFERENCES_KEY, !enable);
+      });
+    });
+  }
+
+  void _listenBackupNowRequests(Future<SharedPreferences> sharedPrefrences) {
+    _backupNowController.stream.listen((data) {
+      _service.authorize();
+      _service.backup(_currentBackupPaths, _currentNodeId);
+
+      sharedPrefrences.then((preferences) {
+        _backupDisabledController
+            .add(preferences.getBool(BACKUP_DISABLED_PREFERENCES_KEY) ?? false);
+      });
+    });
+  }
+
+  _listenBackupPaths(BreezBridge breezLib) {
+    Observable(breezLib.notificationStream).where((event) {
+      return event.type ==
+          NotificationEvent_NotificationType.BACKUP_FILES_AVAILABLE;
+    }).listen((event) {
+      _service.backup(event.data, _currentNodeId);
+      _currentBackupPaths = event.data;
+    });
+  }
+
+  void _listenRestoreRequests(BreezBridge breezLib) {
+    _restoreRequestController.stream.listen((nodeId) {
+      _service.restore(nodeId: nodeId).then((restoreResult) {
+        if (restoreResult is List<dynamic>) {
+          // We got a list of local files to restore from
+          // So let's kick-off lighntinglib
+          breezLib.bootstrap().then((done){
+            getApplicationDocumentsDirectory().then((appDir) {
+              breezLib.copyBreezConfig(appDir.path).then((done) {
+                breezLib.bootstrapFiles(appDir.path, new List<String>.from(restoreResult)).then((done) {
+                  _accountBloc.startLightning();
+                });
+              });
+            });
+          });
+        }
+        else if (restoreResult is bool) {
```

Comment:
Better to stay with on type for the method return type and raise error from the API instead.